critwall/src/main/java/org/opencadc/critwall/Main.java

```diff
@@ -229,8 +237,31 @@ public static void main(String[] args) {
                 "org.postgresql.Driver",
                 dbUrl);
 
-            FileSync doit = new FileSync(daoConfig, cc, localStorage, locatorService, bucketSel, nthreads);
-            doit.run();
+            final Subject subject = SSLUtil.createSubject(new File(CERTIFICATE_FILE_LOCATION));
```

Comment:
Two things here:
1. The application should work without a certificate so this needs to check if the file exists before creating a Subject.
2. The FileSync ctor should not do any remote calls that require auth, so only the `run()` needs to be inside the Subject.doAs. It's bad practice to have setup (ctors) require auth so we want to find and fix that kind of thing.
3. There is a standard wrapper (RunnableAction) you can use to execute a Runnable inside a Subject.doAs